addons/survey/static/src/js/survey_session_manage.js

```diff
@@ -183,15 +182,20 @@ publicWidget.registry.SurveySessionManage = publicWidget.Widget.extend({
 
         var screenToDisplay = this._getPreviousScreen();
 
+        if (this.isFirstQuestion && screenToDisplay === 'previousQuestion') {
+            return;
+        }
+
         if (screenToDisplay === 'question') {
             this._setShowInputs(false);
-            this.$('.o_survey_session_navigation_previous').addClass('d-none');
         } else if (screenToDisplay === 'userInputs') {
             this._setShowAnswers(false);
             // resume refreshing answers if necessary
             if (!this.resultsRefreshInterval) {
                 this.resultsRefreshInterval = setInterval(this._refreshResults.bind(this), 2000);
             }
+        } else if (this.currentScreen === 'question' || screenToDisplay === 'previousQuestion') {  // If go back to previous question
```

Comment:
One last (last last) thing.
To keep it as easy to read as possible, I would only trust "screenToDisplay" here, currentScreen itself is tested extensively by the "_getPreviousScreen" method.

In other words:
```js
...
} else if (screenToDisplay === 'previousQuestion') {
    if (this.isFirstQuestion) {
        return; // nothing to go back to, we're on the first question
    }
    this._nextQuestion(true);
}
```

I was also thinking about putting that condition last, since the conditions are sorted "ascending" so question, then inputs, then results, then we would have your special previous question case.

So we keep the "centralized" if/else switch to test the screen to display in one place and act accordingly. 
What do you think?

Reply:
looks better indeed